geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneQueriesReindexDUnitTest.java

```diff
@@ -90,6 +93,65 @@ public void dropAndRecreateIndex(RegionTestableType regionTestType) throws Excep
     executeTextSearch(accessor);
   }
 
+  @Test
+  @Parameters(method = "getListOfRegionTestTypes")
+  public void dropAndRecreateIndexWithDifferentFieldsShouldFail(RegionTestableType regionTestType)
```

Comment:
Do we want a test where it should just fail on the first time as well?

Are there other test scenarios we are interested in?